lib/babe/epoch.go

```diff
@@ -222,11 +223,26 @@ func (b *Service) incrementEpoch() (uint64, error) {
 // if it is not authorised.
 // output = return[0:32]; proof = return[32:96]
 func (b *Service) runLottery(slot, epoch uint64, epochData *epochData) (*VrfOutputAndProof, error) {
-	return claimPrimarySlot(
+	proof, err := claimPrimarySlot(
 		epochData.randomness,
 		slot,
 		epoch,
 		epochData.threshold,
 		b.keypair,
 	)
+	if err == nil {
+		b.slotToIfPrimary[slot] = true
```

Comment:
remove this from the `Service` - I would actually put this into two separate functions, `runPrimaryLottery` and `runSecondaryLottery`. then, in `epochHandler` you could separate the current `slotToProof` into `primarySlotToProof` and `secondarySlotToProof` and store the proofs there in separate maps. this will also make it easier in the future if we implement secondary VRF

Reply:
using pre runtime digests now, so don't need to use maps for proofs and if primary.